internal/provider/provider.go

```diff
@@ -65,12 +71,16 @@ func (p *Provider) Configure(ctx context.Context, req provider.ConfigureRequest,
 	opts := &ProviderOpts{
 		// This is only for testing, so we can inject provider config
 		repositories: p.repositories,
+		packages:     p.packages,
 		keyring:      p.keyring,
 		archs:        p.archs,
 	}
 	if len(data.Repositories) > 0 {
 		opts.repositories = data.Repositories
 	}
+	if len(data.Packages) > 0 {
+		opts.packages = data.Packages
+	}
 	if len(data.Keyring) > 0 {
 		opts.keyring = data.Keyring
 	}
```

Comment:
These all just get `append`ed anyway, so it doesn't matter if they are empty. Same for archs but GitHub wouldn't let me comment on it to make the suggestion.

Reply:
I think archs is the only one that is not appended, but we can probably make all of these `append()` in the initialization. 🤔